core/src/main/kotlin/fr/sncf/osrd/api/api_v2/pathfinding/PathfindingBlocksEndpointV2.kt

```diff
@@ -138,63 +147,192 @@ private fun computePaths(
+    val pathProps =
+        makePathProps(infra.rawInfra, infra.blockInfra, pathRanges.map { it.edge }, Offset.zero())
+    val blockedRanges = getBlockedRanges(infra, pathRanges, gaugeConstraint)
+
+    // Split response on different gauges (even if no value is associated)
+    val pathGauge = pathProps.getLoadingGauge()
+    val filteredPathGauge = filterIntersection(pathGauge, blockedRanges)
+    filteredPathGauge.shiftPositions(-travelledPathOffset)
+    return filteredPathGauge.map { Pathfinding.Range(Offset(it.lower), Offset(it.upper)) }
+}
+
+private fun getSignalingSystemBlockedRangeValues(
+    infra: FullInfra,
+    pathRanges: List<Pathfinding.EdgeRange<BlockId, Block>>,
+    signalingSystemConstraint: SignalingSystemConstraints?
+): List<RangeValue<String>> {
+    if (signalingSystemConstraint == null) {
+        return listOf()
+    }
+
+    val travelledPathOffset = pathRanges.first().start.distance
+    val blockList = pathRanges.map { it.edge }
+    val blockedRanges = getBlockedRanges(infra, pathRanges, signalingSystemConstraint)
+
+    val pathSignalingSystem = getPathSignalingSystems(infra, blockList)
+    val filteredPathSignalingSystem = filterIntersection(pathSignalingSystem, blockedRanges)
+    filteredPathSignalingSystem.shiftPositions(-travelledPathOffset)
+    return filteredPathSignalingSystem.map {
+        RangeValue(Pathfinding.Range(Offset(it.lower), Offset(it.upper)), it.value)
+    }
+}
+
+private fun getBlockedRanges(
+    infra: FullInfra,
+    pathRanges: List<Pathfinding.EdgeRange<BlockId, Block>>,
+    currentConstraint: PathfindingConstraint<Block>
+): DistanceRangeMap<Boolean> {
+    val blockList = pathRanges.map { it.edge }
+    val blockedRanges = distanceRangeMapOf<Boolean>()
+    var startBlockPathOffset = Distance.ZERO
+    for (block in blockList) {
+        currentConstraint.apply(block).map {
+            blockedRanges.put(
+                startBlockPathOffset + it.start.distance,
+                startBlockPathOffset + it.end.distance,
+                true
+            )
+        }
+        startBlockPathOffset += infra.blockInfra.getBlockLength(block).distance
+    }
+    blockedRanges.truncate(
+        pathRanges.first().start.distance,
+        startBlockPathOffset - infra.blockInfra.getBlockLength(blockList.last()).distance +
+            pathRanges.last().end.distance
+    )
+    return blockedRanges
+}
+
+private fun getPathSignalingSystems(
+    infra: FullInfra,
+    blockList: List<BlockId>
+): DistanceRangeMap<String> {
+    val pathSignalingSystem = distanceRangeMapOf<String>()
+    var startBlockPathOffset = Distance.ZERO
+    for (block in blockList) {
+        val blockLength = infra.blockInfra.getBlockLength(block).distance
+        val blockSignalingSystemIdx = infra.blockInfra.getBlockSignalingSystem(block)
+        val blockSignalingSystemName =
+            infra.signalingSimulator.sigModuleManager.getName(blockSignalingSystemIdx)
```

Comment:
@Khoyo: not totally confident that this (conbined with the add of `getName` to the interface https://github.com/OpenRailAssociation/osrd/pull/7953/files#diff-481923950eaa8c032e916301d46efc041482a6364a59cfc1bab17d9019c33930R54) is the best way to get a readable name for signaling system.

Reply:
LGTM ;)